### PR TITLE
LPS-107862 Script handling updates and add tests

### PR DESCRIPTION
**Pull request by christopherbryanboyd**

### CODE REVIEW

**Comment by christopherbryanboyd** [comment]
ci:test:gradle-plugins

**Comment by christopherbryanboyd** [comment]
ci:stop

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/build.gradle [hunk 1]
```diff
@@ -0,0 +1,12 @@
+task runGradleTest
+
+runGradleTest {
+	//dependsOn 'removeDockerContainer'
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/build.gradle]: Remove this line,

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+buildscript {
+	dependencies {
+		classpath fileTree(dir: pluginClasspathDir, excludes: ["biz.aQute.bnd-3.5.0.jar", "com.liferay.gradle.util-1.0.33.jar"], include: "*.jar")
+	}
+}
+
+apply plugin: "com.liferay.workspace"
+
+gradle.liferayWorkspace {
+	dockerImageId = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase()
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]: Lets use hostname + test name to further differeientiate

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			public void write(String line) {
+				if (line.contains("org.apache.catalina.startup.Catalina")) {
+					scriptFired = true
+				}
+				gradle.rootProject.print line
+			}
+		}
+
+		doLast {
+			gradle.rootProject.println "DoLast!"
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]: no need for this.

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainerWithScript/build.gradle [hunk 4]
```diff
@@ -0,0 +1,12 @@
+task runGradleTest
+
+runGradleTest {
+	//dependsOn 'removeDockerContainer'
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainerWithScript/build.gradle]: remove

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	gradle.rootProject.tasks.register("assertScript", DockerLogsContainer) {
+		dependsOn gradle.rootProject.tasks.startDockerContainer
+		finalizedBy gradle.rootProject.tasks.cleanDockerImage
+		tailAll = true
+		follow = false
+		targetContainerId {
+			gradle.liferayWorkspace.dockerContainerId
+		}
+
+		boolean scriptFired = false
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]: this is not a good descriptive variable

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+apply plugin: "com.liferay.workspace"
+
+gradle.liferayWorkspace {
+	dockerImageId = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase()
+	dockerContainerId = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase()
+}
+
+gradle.projectsEvaluated {
+	gradle.rootProject.tasks.register("assertScript", DockerLogsContainer) {
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]: we access gradle.rootProject over and over, so lets assign it to a var

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+			int retries = 0
+
+			while (!scriptFired && retries < 50) {
+				Thread.sleep(100)
+				retries++
+				start()
+			}
+
+			if (!scriptFired) {
+				throw new GradleException("Script did not fire.")
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]: Lets use better description

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle (hunk 6 above)
Comment (christopherbryanboyd) [reply modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]: Gradle won't allow us to assign `gradle.rootProject` to a script variable due to an error,
Caused by: java.lang.IllegalStateException: The root project is not yet available for build.

**Comment by christopherbryanboyd** [comment]
ci:test:gradle-plugins

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/providedModules/build.gradle [hunk 8]
```diff
@@ -9,7 +9,7 @@ runGradleTest {
 	dependsOn initBundle
 
 	doLast {
-		configurations.providedModules.dependencies.each {
+		configurations.compileOnlyModules.dependencies.each {
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/providedModules/build.gradle]: why did you change this?

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle (hunk 6 above)
Comment (gamerson) [reply modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]: So inside the `projectsEvaluated` block, declare the root project, so it doesn't have to be repeated

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/build.gradle [hunk 9]
```diff
@@ -0,0 +1,5 @@
+task runGradleTest
+
+runGradleTest {
+	dependsOn "assertScript"
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/build.gradle]: lets call this assertLog

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/providedModules/build.gradle (hunk 8 above)
Comment (christopherbryanboyd) [reply modules/sdk/gradle-plugins-workspace/src/gradleTest/providedModules/build.gradle]: I didn't change this explicitly, I suspect it was done by source formatter.

Comment (gamerson) [reply modules/sdk/gradle-plugins-workspace/src/gradleTest/providedModules/build.gradle]: got it... so change it back and add exclusion rule if we need. thanks.

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainerWithScript/settings.gradle [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import com.bmuschko.gradle.docker.tasks.container.DockerLogsContainer
+
+import java.net.InetAddress;
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainerWithScript/settings.gradle]: Remove `;`

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainerWithScript/settings.gradle [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+		classpath fileTree(dir: pluginClasspathDir, excludes: ["biz.aQute.bnd-3.5.0.jar", "com.liferay.gradle.util-1.0.33.jar"], include: "*.jar")
+	}
+}
+
+apply plugin: "com.liferay.workspace"
+
+def expectedLine = "Hello World"
+
+gradle.liferayWorkspace {
+	dockerImageId = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase() + 'startdockercontainerwithscript'
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainerWithScript/settings.gradle]: lets use more idiomatic groovy
`InetAddress.localHost.canonicalHostName.toLowerCase()`

File: modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+def expectedLine = "org.apache.catalina.startup.Catalina"
+
+gradle.liferayWorkspace {
+	dockerImageId = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase() + 'startdockercontainer'
+	dockerContainerId = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase() + 'startdockercontainer'
+}
+
+gradle.projectsEvaluated {
+	def root = gradle.rootProject
```
Comment (gamerson) [inline modules/sdk/gradle-plugins-workspace/src/gradleTest/startDockerContainer/settings.gradle]:
```suggestion
	Project rootProject = gradle.rootProject
```

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/petershin/liferay-portal/pull/880. [See changes here.](https://github.com/gamerson/liferay-portal/compare/2aea91f945e7~6...2aea91f945e7) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/gamerson/liferay-portal/pull/400. [See changes here.](https://github.com/gamerson/liferay-portal/compare/afcceb5a36e2~6...afcceb5a36e2) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/petershin/liferay-portal/pull/882. [See changes here.](https://github.com/gamerson/liferay-portal/compare/f014f5b15361~8...f014f5b15361) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
